### PR TITLE
Include pipeline id in some pass error scopes

### PR DESCRIPTION
**Pull request by scoopr**

### CODE REVIEW

**Comment by scoopr** [comment]
Motivation was going from 
```
Caused by:
    In a RenderPass
      note: encoder = `frame encoder`
    In a draw_indexed command
    pipeline index format (Uint16) and buffer index format (Uint32) do not match
```
to
```
Caused by:
    In a RenderPass
      note: encoder = `frame encoder`
    In a draw_indexed command
      note: render pipeline = `render pipe`
    pipeline index format (Uint16) and buffer index format (Uint32) do not match
```

File: wgpu-core/src/command/mod.rs [hunk 1]
```diff
@@ -361,19 +361,19 @@ pub enum PassErrorScope {
     #[error("In a set_scissor_rect command")]
     SetScissorRect,
     #[error("In a draw command")]
-    Draw,
+    Draw(Option<id::RenderPipelineId>),
```
Comment (kvark) [inline wgpu-core/src/command/mod.rs:364]: Can we reduce the duplication here? Something like:
```rust
Draw {
  indexed: bool,
  indirect: bool,
  pipeline: Option<id::RenderPipelineId>,
},
```

**Review by kvark — COMMENTED** [review]
Thank you, it's great to see errors improving!

File: wgpu-core/src/command/mod.rs (hunk 1 above)
Comment (scoopr) [reply wgpu-core/src/command/mod.rs:364]: hmm, that would make the type ever so slightly larger ;) 
I'd prefer to punt it to later

Comment (kvark) [reply wgpu-core/src/command/mod.rs:364]: Does it matter how large the error type is? It's not like you'd be juggling it in perf-sensitive contexts

Comment (scoopr) [reply wgpu-core/src/command/mod.rs]: No, that was in jest :)

**Comment by scoopr** [comment]
Merged the variants,
now the error looks like:
```
Caused by:
    In a RenderPass
      note: encoder = `frame encoder`
    In a draw command, indexed:true indirect:false
      note: render pipeline = `render pipe`
    pipeline index format (Uint16) and buffer index format (Uint32) do not match
```

File: wgpu-core/src/command/compute.rs [hunk 2]
```diff
@@ -459,7 +459,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     }
                 }
                 ComputeCommand::DispatchIndirect { buffer_id, offset } => {
-                    let scope = PassErrorScope::DispatchIndirect;
+                    let scope = PassErrorScope::Dispatch{indirect:true, pipeline:state.pipeline.last_state};
```
Comment (kvark) [inline wgpu-core/src/command/compute.rs:462]: did you run `cargo fmt` on this?

File: wgpu-core/src/command/render.rs [hunk 3]
```diff
@@ -1619,9 +1627,18 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                         indexed,
                     } => {
                         let scope = if indexed {
-                            PassErrorScope::DrawIndexedIndirect
+                            PassErrorScope::Draw {
+                                indexed: true,
```
Comment (kvark) [inline wgpu-core/src/command/render.rs:1631]: now we can merge those two branches into one:
```rust
let scope = PassErrorScope::Draw {
  indexed,
  indirect: true,
  pipeline: state.pipeline.last_state,
};
```

**Review by kvark — COMMENTED** [review]
A few nits, otherwise looks great!

File: wgpu-core/src/command/compute.rs (hunk 2 above)
Comment (scoopr) [reply wgpu-core/src/command/compute.rs]: I did, but I guess I noticed the dispatch stuff after that, oops.

File: wgpu-core/src/command/render.rs (hunk 3 above)
Comment (scoopr) [reply wgpu-core/src/command/render.rs]: oh duh, of course :)

**Review by kvark — APPROVED** [review]
Great!
bors r+

**Pull request merged**